src/code/rand.c

```diff
@@ -88,12 +98,22 @@ void Rand_Seed(u32 seed) {
  * @note Original name: fqrand
  */
 f32 Rand_ZeroOne(void) {
+#if RAND_VERSION == RAND_N64
+    fu v;
+    f32 vf;
+
+    v.i = (Rand_Next() & 0x007FFFFF) | 0x3F800000;
```

Comment:
This should be documented as a bug. `Rand_Next() & 0x007FFFFF` grabs the least significant 23 bits, which reduces the repeat sequence of the RNG from 2^32 down to 2^23

Reply:
Isn't it simply a "not-ideal implementation", so not a bug
It's not like there are strict specs to the rand system and things fall apart due to this

Reply:
well it seems like they considered it a bug themselves did fix it for GC

Reply:
So I took a look into what Numerical Recipes in C had to say. Chapter 7.1 covers uniform deviates, and covers a few implementations for generating one for the range 0 to 1 exclusive. One of those functions is called `ranqd2` (random quick and dirty 2), and is essentially identical to the N64 implementation of this function. The book does bring up that the float number being constructed of the low-order 23 bit s is "not ideal", but I imagine it was done this way without shifting because back then, right shifting by 9 was not always a single 1-cycle opcode like it is in MIPS. So I guess on that front Dragorn is correct in that it's not strictly a bug.

I do think that a comment about the shorter/worse RNG period is worth mentioning though

Reply:
I'm ok adding a comment, though I think the one on the other rand version is enough, but idk what to write

Reply:
This absolutely needs a comment imo. Here's my suggestion:
`// Note this samples the lower 23 bits, effectively reducing the LCG period from 2^32 to 2^23. This was fixed in Gamecube versions and Majora's Mask.`